x-pack/platform/plugins/private/product_intercept_trigger/server/core.ts

```diff
@@ -0,0 +1,232 @@
+  taskManager: TaskManagerStartContract;
+}
+
+interface ProductInterceptTaskState {
+  runs: number;
+  firstScheduledAt: ReturnType<Date['toISOString']>;
+}
+
+interface ProductInterceptTriggerStatusContext extends RequestHandlerContext {
+  triggerInfo: Promise<
+    | (ProductInterceptTaskState & {
+        interval: IntervalSchedule['interval'];
+        runAt: ConcreteTaskInstance['runAt'];
+      })
+    | null
+  >;
+}
+
+export class ProductInterceptTriggerCore {
+  private readonly core: CoreSetup;
+  private readonly logger: Logger;
+  private readonly taskManagerSetup: TaskManagerSetupContract;
+  private taskManager?: TaskManagerStartContract;
+  private readonly isServerless: boolean;
+
+  private readonly taskType = 'productInterceptDialogTrigger';
+  // define a scope so we might be able to use this to track if we have already scheduled a task in time past
+  private readonly taskScope = 'productInterceptTrigger';
+
+  constructor(
+    private readonly ctx: PluginInitializerContext<unknown>,
+    { core, logger, taskManager }: ProductInterceptTriggerCoreInitDeps
+  ) {
+    this.core = core;
+    this.logger = logger;
+    this.taskManagerSetup = taskManager;
+    this.isServerless = this.ctx.env.packageInfo.buildFlavor === 'serverless';
+
+    this.registerTrigger.call(this);
+    this.registerRoutes.call(this);
+  }
+
+  registerTrigger() {
+    this.taskManagerSetup.registerTaskDefinitions({
+      [this.taskType]: {
+        title: 'Product intercept dialog trigger',
+        description: 'Task that triggers the product intercept dialog',
+        // To ensure the validity of task state during read and write operations, utilize the stateSchemaByVersion configuration. This functionality validates the state before executing a task. Make sure to define the schema property using the @kbn/config-schema plugin, specifically as an ObjectType (schema.object) at the top level.
+        stateSchemaByVersion: {
+          1: {
+            schema: schema.object({
+              runs: schema.number(),
+              firstScheduledAt: schema.string(),
+            }),
+            up: (state) => {
+              return {
+                runs: state.runs || 0,
+                firstScheduledAt: state.firstScheduledAt || new Date().toISOString(),
+              };
+            },
+          },
+        },
+        createTaskRunner: (context) => {
+          return {
+            run: async () => {
+              this.logger.debug('Product intercept trigger ticker is called!');
+
+              return {
+                // updating the state value provides the basis that allows us to infer if the user
+                // interacted with the dialog on the last run if there was one on the client
+                state: {
+                  runs: context.taskInstance.state.runs + 1,
+                  firstScheduledAt: context.taskInstance.state.firstScheduledAt,
+                },
+              };
```

Comment:
Besides returning an updated state here, we can leverage the `shouldDeleteTask` option to provide the ability to remove an already configured trigger task that depends on some config we provide that the user can opt in to.